### PR TITLE
Add notification settings

### PR DESCRIPTION
**Pull request by birneamstiel**
Is connected to: #437
issue: #301

### CODE REVIEW

File: app/views/users/notifications.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<div class="page-header">
+  <h1><%= t 'view.user.notifications.settings' %></h1>
+</div>
+<%= form_for(@user, url: user_path(@user), method: :patch) do |f| %>
+  <div class="form-group">
+    <div class="col-lg-offset-2 col-lg-10">
+      <%= f.check_box :team_notifications_enabled %>
```
Comment (bakoe) [inline app/views/users/notifications.html.erb]: My personal preference would be to have the `filled-in` class on these checkboxes which would make them look like this:

ohne titel 2

File: app/views/users/notifications.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      <%= f.label :team_notifications_enabled, t('view.user.notifications.enable_team_notifications') %> <br>
+
+    </div>
+  </div>
+
+  <div class="form-group">
+    <div class="col-lg-offset-2 col-lg-10">
+      <%= f.check_box :event_notifications_enabled %>
```
Comment (bakoe) [inline app/views/users/notifications.html.erb]: See comment to line 7

File: config/locales/de.yml [hunk 3]
```diff
@@ -230,7 +230,12 @@ de:
       edit_profile: "Profilinformationen bearbeiten"
       update_profile: "Profil aktualisieren"
       favourite_sports_placeholder: "Fußball, Basketball, Tischkicker, ..."
-    
+      notifications:
+            settings: "Benachrichtigungseinstellungen"
+            enable_team_notifications: "Team Benachrichtigungen erhalten"
```
Comment (bakoe) [inline config/locales/de.yml]: My personal preference would be “Team-Benachrichtigungen erhalten” (with dash)

File: config/locales/de.yml [hunk 4]
```diff
@@ -230,7 +230,12 @@ de:
...
+            enable_event_notifications: "Event Benachrichtigung erhalten"
```
Comment (bakoe) [inline config/locales/de.yml]: equivalent to comment to line 235

File: spec/controllers/users_controller_spec.rb [hunk 5]
```diff
@@ -318,4 +318,20 @@
...
+      it 'should show notifications settings page successfully' do
+        sign_in @user
+        get :notifications, params: { id: @user.to_param }
+        expect(response).to be_success
+      end
+    end
+    context 'given no logged in user' do
+      it 'should deny access' do
+        get :notifications, params: { id: @user.to_param }
+        expect(response).to be_unauthorized
```
Comment (bakoe) [inline spec/controllers/users_controller_spec.rb:332]: 👍

**Review by bakoe — COMMENTED** [review]
LGTM code-wise! Can confirm it's working in the Browser. Only some minor complaints/hints regarding design and wording

**Comment by bakoe** [comment]
@birneamstiel Please make sure to integrate the latest changes from `dev` into your branch. Pay attention to the now obsolete mocked notification settings getters from `user.rb` (which **must** be removed now):
```ruby
  def has_event_notifications_enabled?
    true
  end

  def has_team_notifications_enabled?
    true
  end
```
and to the mocked setting of these settings in the tests for TeamMailer, EventMailer and MatchMailer:
```ruby
allow(user).to receive(:has_team_notifications_enabled?).and_return(true)
```
which could be replaced by something like 
```ruby
user.team_notifications_enabled = true
```
(make sure to catch all occurrences from TeamMailer, EventMailer and MatchMailer tests!)

**Comment by bakoe** [comment]
ohne titel 2

@birneamstiel Also, I'm not quite sure whether „Aktualisiere Spieler“ is the right label for the button – maybe something like “Benachrichtigungseinstellungen aktualisieren” or simply “Aktualisieren” or even “Speichern” would be better?

**Comment by birneamstiel** [comment]
@bakoe I've merged your changes into this branch, so maybe you could quickly checkout this branch and test your logic (if it's still working)? But I think we're good to merge :-)

**Review by bakoe — APPROVED** [review]
Have tested in Browser and can confirm it's working; LGTM code-wise as well!

**Pull request merged**